include/lsst/daf/base/DateTime.h

```diff
@@ -57,19 +59,27 @@ namespace base {
 
 class DateTime {
 public:
-    enum Timescale { TAI, UTC, TT };
-    enum DateSystem { JD, MJD, EPOCH };
+    enum DateSystem { JD=0, MJD, EPOCH };
+    enum Timescale { TAI=5, UTC, TT };  // use values that do not overlap DateSystem
+                                        // to avoid confusing one for the other in Python
```

Comment:
Changing the start definitely fixes the problem. I added a unit test that proves it.

Normally I'm keen on class enums, but in this case the enum is part of DateTime (e.g. DateTime::TAI in C++, DateTime.TAI in Python), which feels very natural to me and I would prefer not to change it. Especially since that would require an RFC.

Note that SWIG's support for class enums is rather lame: the enum name in Python is ClassName_ENUMNAME. I don't know if the Python layer for class enums is also type safe.